components/chart/src/selection-area/selection-area.ts

```diff
@@ -724,14 +724,20 @@ export class DtChartSelectionArea implements AfterContentInit, OnDestroy {
     // –––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––
     // Decide weather to show the range or a timestamp according to a click or a drag.
     // On a mousedown the range and the timestamp have to be hidden.
+    //
+    // The pipe delay is used in case that both the start and hide listening for the _click$
+    // so the start has to be delayed a little bit that it will be destroyed a few milliseconds
+    // before it gets created again.
     const startShowingTimestamp$ = this._click$.pipe(mapTo(true));
     const startShowingRange$ = this._drag$.pipe(mapTo(true));
-    let hideTimestampAndRange$ =
-      this._chart._range && !this._chart._timestamp
-        ? EMPTY
-        : touchStartAndMouseDown$.pipe(mapTo(false));
-
-    merge(startShowingRange$, hideTimestampAndRange$)
+    const hideTimestampAndRange$ = ((this._chart._timestamp &&
+    !this._chart._range
+      ? this._click$
+      : this._chart._range && !this._chart._timestamp
+      ? this._chart._range!._closeOverlay.asObservable()
+      : touchStartAndMouseDown$) as Observable<any>).pipe(mapTo(false));
```

Comment:
It's kinda hard to read this way. 
Can we not use nested ternary operators? 
```
let hideTimestampAndRange = touchStartAndMouseDown$;
if (this._chart._timestamp && !this._chart._range) {
  hideTimestampAndRange$ = this._click$;
}
....
```
You get the point